# Descriptive Statistics — Block 2, Week 2
**Aayan Mulani │ Decimal Point Analytics Preparation**

Descriptive statistics are the first thing you compute before any modelling.
They tell you the shape, centre, and spread of your data.
In finance, these numbers reveal how returns behave — are they symmetric?
Are crashes more common than rallies? How fat are the tails?

In [1]:
import yfinance as yf
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats

## 1. Downloading Data
We download Nifty 50 daily closing prices from 2020 to 2025.
We then compute daily percentage returns — this is what we run statistics on.
Raw prices are not stationary (they trend upward), but returns are.
We multiply by 100 to express returns as percentages (e.g. 0.85% instead of 0.0085).

In [5]:
nifty = yf.download('^NSEI', start = '2020-01-01', end = '2025-01-01')['Close']
returns = nifty.pct_change().dropna()*100

print(f'Total Trading Days: {len(returns)}')
print(f'Date Range : {returns.index[0].date()} to {returns.index[-1].date()}')
print(f'\nFirst 5 Returns:')
print(returns.head())

[*********************100%***********************]  1 of 1 completed

Total Trading Days: 1236
Date Range : 2020-01-02 to 2024-12-31

First 5 Returns:
Ticker         ^NSEI
Date                
2020-01-02  0.818389
2020-01-03 -0.452279
2020-01-06 -1.910585
2020-01-07  0.499459
2020-01-08 -0.228994


## 2. Core Descriptive Statistics
We compute 8 key statistics on Nifty daily returns.
Each number tells us something different about the shape and behaviour of returns.

In [10]:
print(f'Mean:     {returns.mean().values[0]:.4f}%')
print(f'Median:   {returns.median().values[0]:.4f}%')
print(f'Std Dev:  {returns.std().values[0]:.4f}%')
print(f'Variance: {returns.var().values[0]:.4f}')
print(f'Skewness: {returns.skew().values[0]:.4f}')
print(f'Kurtosis: {returns.kurtosis().values[0]:.4f}')
print(f'Min:      {returns.min().values[0]:.4f}%')
print(f'Max:      {returns.max().values[0]:.4f}%')

Mean:     0.0610%
Median:   0.1092%
Std Dev:  1.2041%
Variance: 1.4499
Skewness: -1.3945
Kurtosis: 18.9748
Min:      -12.9805%
Max:      8.7632%


## 3. Interpretation

**Mean vs Median:**
Mean is lower than median because there are few very extreme negative days (like the COVID crash) pulling the mean down. The median is unaffected because it only cares about the middle value, not the size of extremes.

**Skewness (-1.39):**
The negative sign shows that data is negatively skewed, that means most of the data points are on right sides but on left sides there are huge negative points making it left tailed data. The main reason for this skewness is covid pandemic

**Kurtosis (18.97):**
The Kurtosis value of 18.97 is extremely higher than the normal one (3). It means that tails are drastically fatter than normal tails and probability of extreme events very higher than a normal distribution can predict.

**Key takeaway:**
From this we can infer that market is highly volatile with extreme events and standard models that assume normality will underestimate the probability of extreme losses.